src/api/app/services/scm_status_reporter.rb

```diff
@@ -36,6 +36,8 @@ def call
     @workflow_run.update_to_fail("Failed to report back to GitHub: #{e.message}") if @workflow_run.present?
   rescue Octokit::Error, Gitlab::Error::Error => e
     rescue_with_handler(e) || raise(e)
+  rescue Faraday::ConnectionFailed
+    @workflow_run.update_to_fail('Failed to report back to GitHub: Could not connect to the GitHub API')
```

Comment:
```
rescue Faraday::ConnectionFailed => e
    @workflow_run.update_to_fail("Failed to report back to GitHub: #{e.message}")
```
Have you looked into the exception message? Is it something meaningful?

Reply:
@rubhanazeem yes, looking at what we received in errbit its meaningful `Failed to open TCP connection to api.github.com:443 (Connection timed out - user specified timeout)`
I show the message that comes from faraday now